crates/pipeline_manager/src/integration_test.rs

```diff
@@ -1612,412 +1610,782 @@ async fn pipeline_start_without_compiling() {
+                    }
+                }
+            },
+            "config_type": "kafka"
+        }),
+        json!({
+            "name": "object2",
+            "description": "Description of object2",
+            "config": {
+                "kafka": {
+                    "bootstrap_servers": [
+                        "abc2"
+                    ],
+                    "options": {
+                        "example2": "value2"
+                    }
+                }
+            },
+            "config_type": "kafka"
+        }),
+        json!({
+            "name": "object3",
+            "description": "Description of object3",
+            "config": {
+                "kafka": {
+                    "bootstrap_servers": [
+                        "abc3"
+                    ],
+                    "options": {
+                        "example3": "value3",
+                        "example4": "value4"
+                    }
+                }
+            },
+            "config_type": "kafka"
+        }),
+        json!({
+            "name": "object3",
+            "description": "Description of object3",
+            "config": {
+                "kafka": {
+                    "bootstrap_servers": [
+                        "abc3"
+                    ],
+                    "options": {
+                        "example3": "value3",
+                        "example4": "value4"
+                    }
+                }
+            },
+            "config_type": "kafka"
+        }),
+        json!({
+            "name": "object4",
+            "description": "Description of object4",
+            "config": {
+                "kafka": {
+                    "bootstrap_servers": [
+                        "abc4"
+                    ],
+                    "options": {}
+                }
+            },
+            "config_type": "kafka"
+        }),
+        json!({
+            "name": "object4",
+            "description": "Description of object4",
+            "config": {
+                "kafka": {
+                    "bootstrap_servers": [
+                        "abc4"
+                    ],
+                    "options": {}
+                }
+            },
+            "config_type": "kafka"
+        }),
+        json!({
+            "name": "object5",
+            "description": "Description of object5",
+            "config": {
+                "does_not_exist": {
+                    "example": "value"
+                }
+            },
+            "config_type": "does_not_exist"
+        }),
+    )
+    .await
+}
+
+/// Tests the basic functionality (create, retrieve, update and delete)
+/// of the particular object type.
+///
+/// It does a mix of GET, PUT, PATCH, POST and DELETE among the endpoints
+/// all the object types share.
+async fn basic_api_test(
+    config: &TestConfig,
+    object_types: &str,
+    id_field: &str,
+    object1_request: Value,
+    mut object1_response: Value,
```

Comment:
This is a strange design for a test method. Why does it need this many arguments for different objects? Instead of say, a vector of objects or object pairs?

The typical way to do these type of API tests is to do table-based testing, where you create an array of request + expected response pairs, and have a test method execute each and compare results.

Reply:
As test part change will likely take more time and is independent.